lib/workflow/factory.js

```diff
@@ -82,6 +82,7 @@ Factory.prototype._create = function (definition, depth, callback) {
         try {
             node = new NodeClass(self.user, definition.params);
         } catch (err) {
+            err.node_id = definition.id;
```

Comment:
This will cover most of the cases when input params are invalid, but this won't solve other cases when something fails later in the factory, for instance when creating a cache table.

Let's ship it and we will work on improving other cases :-).